ml-agents-envs/mlagents/envs/environment.py

```diff
@@ -527,8 +541,43 @@ def _get_state(self, output: UnityRLOutputProto) -> AllBrainInfo:
             _data[brain_name] = BrainInfo.from_agent_proto(
                 self.worker_id, agent_info_list, self.brains[brain_name]
             )
+        self._parse_side_channel_message(output.side_channel)
         return _data
 
+    def _parse_side_channel_message(self, data: bytearray) -> None:
+        offset = 0
+        while offset < len(data):
+            try:
+                channel_type = struct.unpack("i", data[offset : offset + 4])[0]
+                offset = offset + 4
+                message_len = struct.unpack("i", data[offset : offset + 4])[0]
+                offset = offset + 4
+                message_data = data[offset : offset + message_len]
+                offset = offset + message_len
+            except Exception:
+                raise UnityEnvironmentException(
+                    "There was a problem reading a message in a SideChannel. "
+                    + "Please make sure the version of MLAgents in Unity is "
+                    + "compatible with the Python version."
+                )
+            if channel_type in self.side_channels_dict:
+                self.side_channels_dict[channel_type].on_message_received(message_data)
```

Comment:
Same comment as C# side, wrap this in a try/except.

Reply:
Same comment, I think we need to raise these errors as they are for the user that made its own side channels.